ugs-core/src/com/willwinder/universalgcodesender/GrblUtils.java

```diff
@@ -240,12 +241,16 @@ static protected String getViewParserStateCommand(final double version, final Ch
     static protected Capabilities getGrblStatusCapabilities(final double version, final Character letter) {
         Capabilities ret = new Capabilities();
         ret.addCapability(CapabilitiesConstants.JOGGING);
+        ret.addCapability(CapabilitiesConstants.HOMING);
+        ret.addCapability(CapabilitiesConstants.HARD_LIMITS);
+        ret.addCapability(CapabilitiesConstants.SOFT_LIMITS);
```

Comment:
I don't think these are supported pre-v8 so they could be moved into one of the conditions below. Maybe change the conditions so that they are 2 if blocks instead of if/else.

Reply:
I went through the old documentation and homing + hard limits looks like they where developed in 0.8. Soft limits in 0.9. Could this be an option:

```java
    static protected Capabilities getGrblStatusCapabilities(final double version, final Character letter) {
        Capabilities ret = new Capabilities();
        ret.addCapability(CapabilitiesConstants.JOGGING);

        if (version >= 0.8) {
            ret.addCapability(CapabilitiesConstants.HOMING);
            ret.addCapability(CapabilitiesConstants.HARD_LIMITS);
        }

        if (version==0.8 && (letter != null) && (letter >= 'c')) {
            ret.addCapability(GrblCapabilitiesConstants.REAL_TIME);
        }

        if (version >= 0.9) {
            ret.addCapability(GrblCapabilitiesConstants.REAL_TIME);
            ret.addCapability(CapabilitiesConstants.SOFT_LIMITS);
            ret.addCapability(CapabilitiesConstants.SETUP_WIZARD);

        }

        if (version >= 1.1) {
            ret.addCapability(GrblCapabilitiesConstants.V1_FORMAT);
            ret.addCapability(GrblCapabilitiesConstants.HARDWARE_JOGGING);
            ret.addCapability(CapabilitiesConstants.OVERRIDES);
            ret.addCapability(CapabilitiesConstants.CONTINUOUS_JOGGING);
        }

        return ret;
    }
```

Reply:
Yes, I think this looks good. The matrix might get more interesting once GRBL-ARM is done, and may even involve querying GRBL for what capabilities are even enabled.